### DISCUSSION TITLE
how to restart kafka-connect properly

### DISCUSSION THREAD

**Discussion by jack1902** [strimzi/strimzi-kafka-operator#9140] [Q&A]
What is the correct way to restart kafka-connect instances?

I am aware that i can restart the following in the following manner:
- `kafka` - annotate pods with `strimzi.io/manual-rolling-update=true`
- `zookeeper` - annotate pods with `strimzi.io/manual-rolling-update=true`
- `connectors` - https://github.com/strimzi/proposals/blob/main/007-restarting-kafka-connect-connectors-and-tasks.md

But i am unsure how to effectively restart the `kafka-connect` instances as i have previously used `kubectl rollout restart deployment <connect-deployment-name>` but on occasion have observed the newer pod getting killed off by a reconcile being carried out by the operator.

The main reason i am restarting the connect instances is because of outdated sidecars which need updating.

**Comment by scholzj** [comment]
Discussed in [Slack](https://cloud-native.slack.com/archives/CMH3Q3SNP/p1695294051876539) => this is a missing feature which should be added in a similar way as for Kafka brokers for example.

**Comment by jack1902** [comment]
Raised: https://github.com/strimzi/strimzi-kafka-operator/issues/9142

**Reply by scholzj** [reply]
Thanks 👍

**Comment by jack1902** [comment]
Adding a comment here as i always come to this issue whenever i google search this

```
kubectl annotate strimzipodset <cluster_name>-connect strimzi.io/manual-rolling-update="true"
```


https://strimzi.io/docs/operators/latest/deploying#proc-manual-rolling-update-strimzipodset-str
